Mathlib/AlgebraicGeometry/EllipticCurve/Weierstrass.lean

```diff
@@ -1154,6 +1176,21 @@ lemma baseChange_j : (E.baseChange A).j = algebraMap R A E.j := by
   map_simp
 #align elliptic_curve.base_change_j EllipticCurve.baseChange_j
 
+lemma baseChange_injective (h : Function.Injective ↑(algebraMap R A)) :
+    Function.Injective <| baseChange (R := R) (A := A) := fun E E' h1 => by
+  simp only [baseChange, mk.injEq] at h1
+  rcases h1 with ⟨h1, h6⟩
+  simp only [WeierstrassCurve.baseChange, WeierstrassCurve.mk.injEq] at h1
+  rcases h1 with ⟨h1, h2, h3, h4, h5⟩
+  ext
+  · exact h h1
+  · exact h h2
+  · exact h h3
+  · exact h h4
+  · exact h h5
+  · simp only [Units.map, MonoidHom.mk'_apply, Units.mk.injEq] at h6
+    exact h h6.1
```

Comment:
```suggestion
lemma baseChange_injective (h : Function.Injective <| algebraMap R A) :
    Function.Injective <| baseChange (R := R) (A := A) := fun E E' h1 => by
  rcases mk.inj h1 with ⟨h1, h2⟩
  replace h2 := (Units.mk.inj h2).left
  rcases WeierstrassCurve.mk.inj h1 with ⟨_, _, _, _, _⟩
  ext <;> apply_fun _ using h <;> assumption
```

Reply:
Here I'd like to call `WeierstrassCurve.baseChange_injective` but I can't figure out how to tell `ext` don't expand everything.